src/pokemon.c

```diff
@@ -1150,8 +1150,8 @@ void CreateBoxMon(struct BoxPokemon *boxMon, u16 species, u8 level, u8 fixedIV,
                 totalRerolls += I_SHINY_CHARM_ADDITIONAL_ROLLS;
             if (LURE_STEP_COUNT != 0)
                 totalRerolls += 1;
-            if (IsCurrentEncounterFishing())
-                totalRerolls += CalculateChainFishingShinyRolls();
+            if (I_FISHING_CHAIN && gIsFishingEncounter)
+                totalRerolls += (2 * gChainFishingDexNavStreak);
```

Comment:
I feel these two changes still justify their functions, like adding `I_FISHING_CHAIN` to `IsCurrentEncounterFishing` and keeping `CalculateChainFishingShinyRolls` to let users add other multipliers for their streaks, like a charm or a Pokémon ability.

Reply:
Imo including `I_FISHING_CHAIN` in `IsCurrentEncounterFishing` would be misleading with the current function name. From the name i would expect it to return `true` even if chain fishing is disabled.